Zend/zend_object_handlers.c

```diff
@@ -565,18 +565,27 @@ ZEND_API uint32_t *zend_get_property_guard(zend_object *zobj, zend_string *membe
 	zval *zv;
 	uint32_t *ptr;
 
+	if (EXPECTED(EG(guard_context) == 0)) {
+		member = zend_string_copy(member);
+	} else {
+		member = zend_string_init(ZSTR_VAL(member), ZSTR_LEN(member), false);
+		ZSTR_H(member) = zend_string_hash_val(member) + EG(guard_context);
```

Comment:
I don't particularly like having the overhead of a string allocation for every single property hook access in a fiber. It will also fail the trivial str == member comparison and require a full comparison for any access in fiber context. It's probably okay for just magic __get/__set, but I expect property hooks to be quite common.

Can you just allocate 8 bytes more for zend_objects using guards? And store the guard_context there - at least for the case where no nested (or parallel in case of fibers) access happens?

Reply:
>  but I expect property hooks to be quite common.

Hooks don't do guards anymore, so. :)

Reply:
Oh, right. I still think allocating a few more bytes for the guard wouldn't do harm though, i.e. common case.

Reply:
This not only requires a new string allocation for each __get/__set call in a fiber. These strings are also going to be kept in guards hash until the object destruction.

Reply:
That's true. I can try @bwoebi's approach, but that also means increasing the allocation size of all objects with guards, even when they don't use fibers. I would assume that applications using fibers are generally more modern, possibly also making less use of `__get`/`__set`. But of course we can't know for sure.